### PR TITLE
Refactor Sui guide

### PR DESCRIPTION
**Pull request by aditya520**

### CODE REVIEW

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 1]
```diff
@@ -1,82 +1,98 @@
...
 
-```typescript copy
-import { SuiPriceServiceConnection } from "@pythnetwork/pyth-sui-js";
+</Tabs.Tab>
+</Tabs>
 
-const connection = new SuiPriceServiceConnection(
-  "https://hermes-beta.pyth.network"
-); // See Hermes endpoints section below for other endpoints
+Pyth provides a javascript SDK to construct transaction blocks that update price feeds.
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:52]: nit: Pyth also provides

Comment (jayantk) [reply pages/price-feeds/use-real-time-data/sui.mdx:52]: oh also this sentence and the one above should end with a : not a .

**Review by jayantk — COMMENTED** [review]
on the FAQ section, i think the answers to both questions can be inlined in the docs. I don't think this FAQ question corresponds to real questions users asked; I think Leo wrote it by himself anticipating those questions.

I think the "sui rev" question could be a code comment on the Move.toml code snippet. I think the answer can be shortened so that it fits reasonably as a comment

The price object question probably gets inlined as a comment somewhere in the typescript snippet.

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 2]
```diff
@@ -1,82 +1,124 @@
...
+subdir = "target_chains/sui/contracts"
+rev = "sui-contract-mainnet"
+
+[dependencies.Wormhole]
+git = "https://github.com/wormhole-foundation/wormhole.git"
+subdir = "sui/wormhole"
+rev = "sui-upgrade-mainnet"
 
-### Installation
+# Pyth is locked into this specific `rev` because our package depends on Wormhole and is pinned to this version.
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:24]: because the package

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 3]
```diff
@@ -1,82 +1,124 @@
...
+subdir = "target_chains/sui/contracts"
+rev = "sui-contract-testnet"
+
+[dependencies.Wormhole]
+git = "https://github.com/wormhole-foundation/wormhole.git"
+subdir = "sui/wormhole"
+rev = "sui-upgrade-testnet"
 
-If you are using yarn run:
+# Pyth is locked into this specific `rev` because our package depends on Wormhole and is pinned to this version.
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:45]: same comment: our -> the

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 4]
```diff
@@ -1,82 +1,124 @@
...
+subdir = "target_chains/sui/vendor/wormhole_movement_devnet/wormhole"
+rev = "sui-contract-movement-m2-devnet"
 
-Pyth stores prices off-chain to minimize gas fees, which allows us to offer a wider selection of products and faster update times.
-See [Pull Updates](../pythnet-price-feeds/pull-updates) for more information about this approach.
-Typically, to use Pyth prices on chain,
-they must be fetched from the [Hermes API](https://hermes.pyth.network/docs). The `SuiPriceServiceConnection` class can be used to interact with these services,
-providing a way to fetch these prices directly in your code. The following example wraps an existing RPC provider and shows how to obtain
-Pyth prices and submit them to the network:
+# Pyth is locked into this specific `rev` because our package depends on Wormhole and is pinned to this version.
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:66]: here also

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 5]
```diff
@@ -1,82 +1,124 @@
...
+    ```sh copy npm install --save @pythnetwork/pyth-sui-js ```
+  </Tabs.Tab>
+  <Tabs.Tab>```sh copy yarn add @pythnetwork/pyth-sui-js ```</Tabs.Tab>
+</Tabs>
 
-## On-chain prices
+## Write Contract Code
 
-### ⚠️ **_Important Note for Integrators_**
+<Callout type="warning" emoji="⚠️">
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:89]: I think this callout should come after the code snippet.

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 6]
```diff
@@ -1,82 +1,124 @@
...
+Your Sui Move module **should NOT** have a hard-coded call to `pyth::update_single_price_feed.` In other words, a contract should **never call** the Sui Pyth `pyth::update_single_price_feed` entry point. Instead, it should be called directly from client code (e.g., Typescript or Rust).
 
-This is because when a Sui contract is [upgraded](https://docs.sui.io/build/package-upgrades), the new address is different from the original. If your module has a hard-coded call to `pyth::update_single_price_feed` living at a fixed call-site, it may eventually get bricked due to the way Pyth upgrades are implemented. (We only allows users to interact with the most recent package version for security reasons).
+When Sui contracts are [upgraded](<(https://docs.sui.io/build/package-upgrades)>), the address changes, which makes the old address no longer valid. If your module has a hard-coded call to `pyth::update_single_price_feed` living at a fixed call-site, it may eventually get bricked due to how Pyth upgrades are implemented. (We only allow users to interact with the most recent package version for security reasons).
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:93]: no "We" here

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 7]
```diff
@@ -1,82 +1,124 @@
...
 Therefore, you should build a [Sui programmable transaction](https://docs.sui.io/build/prog-trans-ts-sdk) that first updates the price by calling `pyth::update_single_price_feed` at the latest call-site from the client-side and then call a function in your contract that invokes `pyth::get_price` on the `PriceInfoObject` to get the recently updated price.
-You can use `SuiPythClient` to build such transactions and handles all the complexity of updating the price feeds.
+You can use `SuiPythClient` to build such transactions and handle all the complexity of updating the price feeds.
+
+Consult [Fetch Price Updates](../fetch-price-updates) for more information on how to fetch the `pyth_price_update`.
+
+</Callout>
 
-### Update Example
+The code snippet below provides an example of how to update the Pyth price feeds:
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:102]: I think the organization of these code snippets is confusing. Most of our guides have the contract code first, then they don't have the client code at all. In this case, maybe the contract code first, then the client code as a separate section (sort of how we used to have it before?)

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 8]
```diff
@@ -121,98 +163,40 @@ module pyth_example::main {
...
-[dependencies.Sui]
-git = "https://github.com/MystenLabs/sui.git"
-subdir = "crates/sui-framework/packages/sui-framework"
-rev = "041c5f2bae2fe52079e44b70514333532d69f4e6"
-```
-
 ### CLI Example
 
-[This example](./src/examples/SuiRelay.ts) shows how to update prices on an Sui network. It does the following:
+[This example](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/sui/cli) shows how to update prices on a Sui network. It does the following:
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:194]: I think the example should be under Additional Resources

File: pages/price-feeds/use-real-time-data/sui.mdx [hunk 9]
```diff
@@ -1,82 +1,124 @@
...
+  "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace", // ETH/USD price ID
+];
+
+const priceUpdateData = await connection.getPriceFeedsUpdateData(priceIds);
 
-// It is either injected from browser or instantiated in backend via some private key
+// It is either injected from the browser or instantiated in the backend via some private key
 const wallet: SignerWithProvider = getWallet();
-// Get the state ids of the Pyth and Wormhole contracts from
+// Get the state IDs of the Pyth and Wormhole contracts from
```
Comment (jayantk) [inline pages/price-feeds/use-real-time-data/sui.mdx:121]: fill these in with the same mainnet values that we're using above?

**Review by jayantk — APPROVED** [review]

File: pages/price-feeds/use-real-time-data/sui.mdx (hunk 5 above)
Comment (aditya520) [reply pages/price-feeds/use-real-time-data/sui.mdx:89]: This is pretty important piece of information, so I believe the callout should be before the code snippet.

File: pages/price-feeds/use-real-time-data/sui.mdx (hunk 7 above)
Comment (aditya520) [reply pages/price-feeds/use-real-time-data/sui.mdx:102]: I agree, but especially for sui, it is recommended not to update the price from the contract, I believe we keep this flow.

**Pull request merged**